src/core/Microsoft.PowerToys.Settings.UI.UnitTests/ViewModelTests/ColorPicker.cs

```diff
@@ -2,13 +2,73 @@
 // The Microsoft Corporation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Globalization;
+using System.IO;
+using System.Text.Json;
+using Microsoft.PowerToys.Settings.UI.Lib;
+using Microsoft.PowerToys.Settings.UI.Lib.ViewModels;
+using Microsoft.PowerToys.Settings.UI.UnitTests.BackwardsCompatibility;
+using Microsoft.PowerToys.Settings.UI.UnitTests.Mocks;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
 namespace ViewModelTests
 {
     [TestClass]
     public class ColorPicker
     {
+        private const string OriginalModuleName = "ColorPicker";
+
+
+        /// <summary>
+        /// Test if the original settings files were modified.
+        /// </summary>
+        [TestMethod]
+        [DataRow("v0.20.1", "settings.json")]  //Color picker was introduced in .20
+        [DataRow("v0.21.1", "settings.json")]
+        [DataRow("v0.22.0", "settings.json")]
+        public void OriginalFilesModificationTest(string version, string fileName)
+        {
+            //Arrange
+            var mockIOProvider = BackCompatTestProperties.GetModuleIOProvider(version, OriginalModuleName, fileName);
+            var mockSettingsUtils = new SettingsUtils(mockIOProvider.Object);
+            ColorPickerSettings originalSettings = mockSettingsUtils.GetSettings<ColorPickerSettings>(OriginalModuleName);
+
+            var mockGeneralIOProvider = BackCompatTestProperties.GetGeneralSettingsIOProvider(version);
+            var mockGeneralSettingsUtils = new SettingsUtils(mockGeneralIOProvider.Object);
+            GeneralSettings originalGeneralSettings = mockGeneralSettingsUtils.GetSettings<GeneralSettings>();
+            var generalSettingsRepository = new BackCompatTestProperties.MockSettingsRepository<GeneralSettings>(mockGeneralSettingsUtils);
+
+            //Act
+            // Initialise View Model with test Config files
+            ColorPickerViewModel viewModel = new ColorPickerViewModel(mockSettingsUtils, generalSettingsRepository, ColorPickerIsEnabledByDefault_IPC);
+
+            //Assert
+            // Verifiy that the old settings persisted
```

Comment:
I think I'm missing the idea behind the tests. Was it to validate that configurations persist everytime we open powertoys of a specific version or that the configurations of the powertoys persist on version upgrade? The tests seem to be doing the former but shouldn't we also check the later?

Reply:
> I think I'm missing the idea behind the tests. Was it to validate that configurations persist everytime we open powertoys of a specific version or that the configurations of the powertoys persist on version upgrade? The tests seem to be doing the former but shouldn't we also check the later?

The idea was to validate the scenario that the the current code can handle loading any previous save file.  We don't want to assume which version they are upgrading from.   If we support a setting it should be exposed via the view model, if we don't then then it shouldn't crash or prevent any of the settings that exist across pt versions from loading.